### DISCUSSION TITLE
Webrtc flooding lossy tunnel

### DISCUSSION THREAD

**Discussion by uzername123** [bluenviron/mediamtx#3646] [Questions]
### Question

i trying to send video over lossy  lowspeed (5mbit only/50-80ms latency ) channel ,  using webrtc stream (as it udp and have internal FEC)
on sender part - stream just served as rtsp and webrtc

runOnDemand: gst-launch-1.0 ...... ! queue max-size-bytes=1024000 ! videoconvert ! x264enc bitrate=2048 bframes=0 tune=zerolatency speed-preset=ultrafast sliced-threads=true noise-reduction=5000 ! video/x-h264 ! h264parse ! rtspclientsink location=rtsp://localhost:8554/stream

on receiver - using string:

source: whep://$SENDER_IP:8889/stream/whep

so all clients connecting to local host that serving rtsp /rtmp/webrtc/hls /whatever
remote unit starting to send stream, but when appearing significant losses and delays in channel - seems mediamtx starting to buffering packets that are late because of delay/lossess, and sending it anyway, so it starting to using full channel bandwidth - easily use 5-10megabits, but stream is CBR 2mbit, how to fix that (discard packets that are too late? )?
also - sender printing such string when traffic increasing to max:
 WAR [WebRTC] [session ffbc9fd8] write queue is full

**Comment by aler9** [comment]
Hello, MediaMTX has a fixed-size buffer that it uses for compensating network fluctuations. The size can be tuned by using `writeQueueSize`. Nonetheless, it surely does not hold data in the megabyte range, and judging from the log line you provided, it's already discarding packets.

Therefore, delays are most certainly not caused by the servers, but are caused by buffers inside the clients. If you need further help, provide exact instructions on how to replicate, including the client software and version.
